plugins/authentication/request-error.js

```diff
@@ -27,6 +27,7 @@ function authenticationRequestError (state, error, options) {
       const newOptions = Object.assign(options, {
         headers: Object.assign({ 'x-github-otp': oneTimePassword }, options.headers)
       })
+      state.otp = oneTimePassword
       return state.octokit.request(newOptions)
```

Comment:
I _think_ that with the addition to `before-request.js`, we no longer need to set the `x-github-otp` header here, we could remove lines 27-29 and just do `return state.octokit.request(options)`. I didn’t test it though.

I’ll be offline now and back tomorrow

Reply:
You're right, it makes sense. Refactored in https://github.com/octokit/rest.js/pull/1281/commits/e635dc9d4b2df22ccd0e25f61a6eb9703a255854.

Will look at the other changes tomorrow. Have a good night!